zilencer/views.py

```diff
@@ -500,6 +502,36 @@ def batch_create_table_data(
         row_objects = row_objects[BATCH_SIZE:]
 
 
+def update_remote_realm_data_for_server(
+    server: RemoteZulipServer, server_realms_info: List[Dict[str, Any]]
+) -> None:
+    uuids = [realm["uuid"] for realm in server_realms_info]
+    already_registered_uuids = [
+        str(uuid)
+        for uuid in RemoteRealm.objects.filter(uuid__in=uuids, server=server).values_list(
+            "uuid", flat=True
+        )
+    ]
+
+    new_remote_realms = [
+        RemoteRealm(
+            server=server,
+            uuid=realm["uuid"],
+            uuid_owner_secret=realm["uuid_owner_secret"],
+            host=realm["host"],
+            realm_deactivated=realm["deactivated"],
+            realm_date_created=timestamp_to_datetime(realm["date_created"]),
+        )
+        for realm in server_realms_info
+        if realm["uuid"] not in already_registered_uuids
+    ]
+
+    try:
+        RemoteRealm.objects.bulk_create(new_remote_realms)
+    except IntegrityError:
+        raise JsonableError(_("Duplicate RemoteRealm registration detected."))
```

Comment:
This can probably be just "Duplicate registration detected." to evade the capitalization linter.